Civi/Payment/PropertyBag.php

```diff
@@ -258,13 +271,18 @@ protected function handleLegacyPropNames($prop, $silent = FALSE) {
     if ($newName === NULL && substr($prop, -2) === '-' . \CRM_Core_BAO_LocationType::getBilling()
       && isset(static::$propMap[substr($prop, 0, -2)])
     ) {
-      $newName = substr($prop, 0, -2);
+      $billingAddressProp = substr($prop, 0, -2);
+      $newName = static::$propMap[$billingAddressProp] ?? NULL;
+      if ($newName === TRUE) {
+        // Good, modern name.
+        return $billingAddressProp;
+      }
```

Comment:
For billing address fields make sure we actually return the "new"/propertyBag name if we have a legacy property.

To test eg.  
```
$propertyBag->setBillingCity('Rome');
$city = $propertyBag->getBillingCity();
$city = $propertyBag['billing_city-5'];
```

Reply:
Merged via #21583